apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SpannerDateTests.cs

```diff
@@ -0,0 +1,193 @@
+        [InlineData("9999-12-31")]
+        [InlineData("2022-04-18")]
+        [InlineData("2020-02-29")]
+        public void ParseToStringRoundTrip(string input)
+        {
+            SpannerDate date = SpannerDate.Parse(input);
+            Assert.Equal(input, date.ToString());
+
+            // Check that TryParse works too.
+            Assert.True(SpannerDate.TryParse(input, out var date2));
+            Assert.Equal(date, date2);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("       ")]
+        [InlineData("ABCDEFGH")]
+        [InlineData("pp/qq/rrss")]
+        [InlineData("12-34-5678")]
+        [InlineData("0000-01-01")]
+        [InlineData("9999-12-32")]
+        [InlineData("0000/01/01")]
+        [InlineData("9999/12/32")]
+        [InlineData("2020-00-01")]
+        [InlineData("2020-13-13")]
+        [InlineData("2020-01-00")]
+        [InlineData("2020-01-32")]
+        [InlineData("2020-02-30")]
+        [InlineData("12/31/9999")]
+        [InlineData("0001/01/01")]
+        [InlineData("2022/04/20")]
+        public void Parse_Invalid(string input)
+        {
+            Assert.Throws<FormatException>(() => SpannerDate.Parse(input));
+            bool success = SpannerDate.TryParse(input, out SpannerDate date2);
+            Assert.Equal(default(SpannerDate), date2);
+            Assert.False(success);
+        }
+
+        [Theory]
+        [InlineData("0001-01-01","0001-01-02")]
+        [InlineData("9999-12-30", "9999-12-31")]
+        [InlineData("0001-01-01", "2022-04-18")]
+        [InlineData("2022-04-18", "9999-12-31")]
+        public void Inequality(string input1, string input2)
+        {
+            var date1 = SpannerDate.Parse(input1);
+            var date2 = SpannerDate.Parse(input2);
+            // Value of input1 is always less than input2.
+            // So, we assert that input1 is not equal to input2 and
+            // input1 is less than input2 and
+            // input2 is greater than input1.
+            Assert.NotEqual(date1, date2);
+            Assert.False(date1 == date2);
+            Assert.False(date1.CompareTo(date2) == 0);
+            Assert.False(date2.CompareTo(date1) == 0);
+            Assert.False(date1.GetHashCode() == date2.GetHashCode());
+            Assert.True(date1 != date2);
+            Assert.True(!date1.Equals(date2));
+            Assert.True(!date2.Equals(date1));
+            Assert.True(date1 < date2);
+            Assert.True(date1 <= date2);
+            Assert.True(date1.CompareTo(date2) < 0);
+            Assert.True(date2 > date1);
+            Assert.True(date2 >= date1);
+            Assert.True(date2.CompareTo(date1) > 0);
+        }
+
+        [Theory]
+        [InlineData("0001-01-01", "0001-01-01")]
+        [InlineData("9999-12-31", "9999-12-31")]
+        [InlineData("2022-04-18", "2022-04-18")]
+        [InlineData("2000-02-28", "2000-02-28")]
+        public void Equality(string input1, string input2)
```

Comment:
There's no point in having two inputs when they're the same for every test. Just have a single parameter and parse it twice.